### PR TITLE
Document EnvoyProxy priority problem

### PR DESCRIPTION
**Pull request by frankbu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/ops/common-problems/network-issues/index.md [hunk 1]
```diff
@@ -619,3 +619,41 @@ Most cloud load balancers will not forward the SNI, so if you are terminating TL
...
+error log to indicate that this filter has not been added to the chain.
+
+This is particularly problematic when matching telemetry filters, like `istio.stats`, because they are version
+specific (i.e., they include the `proxyVersion` field in their match criteria) and therefore will be replaced
+by newer ones when upgrading Istio. As a result, an `EnvoyFilter` like the one above may intially be working
+perfectly but after upgrading Istio to a newer version it will no longer be included in the network filter
+chain of the sidecars.
+
+To avoid this issue, you can either change the operation to one that does not depend on the presence of
+another filter (e.g., `INSTERT_FIRST`), or set an exlicit priority in the `EnvoyFilter` to override the
```
Comment (ericvn) [inline content/en/docs/ops/common-problems/network-issues/index.md:657]:
```suggestion
another filter (e.g., `INSERT_FIRST`), or set an exlicit priority in the `EnvoyFilter` to override the
```

File: content/en/docs/ops/common-problems/network-issues/index.md [hunk 2]
```diff
@@ -619,3 +619,41 @@ Most cloud load balancers will not forward the SNI, so if you are terminating TL
...
+default creation time-based ordering. For example, adding `priorty: 10` to the above filter will ensure
```
Comment (ericvn) [inline content/en/docs/ops/common-problems/network-issues/index.md:658]:
```suggestion
default creation time-based ordering. For example, adding `priority: 10` to the above filter will ensure
```

File: content/en/docs/ops/common-problems/network-issues/index.md [hunk 3]
```diff
@@ -619,3 +619,41 @@ Most cloud load balancers will not forward the SNI, so if you are terminating TL
...
+      value:
+        ...
+{{< /text >}}
+
+To work properly, this filter configuration depends on the `istio.stats` filter having an older creation time
+than it. Otherwise, the `INSERT_BEFORE` operation will be silently ignored. There will be nothing in the
+error log to indicate that this filter has not been added to the chain.
+
+This is particularly problematic when matching telemetry filters, like `istio.stats`, because they are version
+specific (i.e., they include the `proxyVersion` field in their match criteria) and therefore will be replaced
```
Comment (howardjohn) [inline content/en/docs/ops/common-problems/network-issues/index.md:651]: just a heads up this will change in the future

**Review by howardjohn — APPROVED** [review]

File: content/en/docs/ops/common-problems/network-issues/index.md [hunk 4]
```diff
@@ -619,3 +619,41 @@ Most cloud load balancers will not forward the SNI, so if you are terminating TL
 - Disable SNI matching in the `Gateway` by setting the hosts field to `*`
 
 A common symptom of this is for the load balancer health checks to succeed while real traffic fails.
+
+## Unchanged envoy filter configuration suddenly stops working
```
Comment (craigbox) [inline content/en/docs/ops/common-problems/network-issues/index.md:623]: "Envoy filter" or "`EnvoyFilter`"

**Review by craigbox — CHANGES_REQUESTED** [review]

**Review by craigbox — APPROVED** [review]

**Pull request merged**